grpc/dial.go

```diff
@@ -7,9 +7,15 @@ import (
 
 	"github.com/edaniels/golog"
 	"go.viam.com/utils/rpc"
+
+	"go.viam.com/rdk/utils/contextutils"
 )
 
-// Dial dials a gRPC server.
+// defaultDialTimeout is the default timeout for dialing a robot.
+var defaultDialTimeout = 20 * time.Second
+
+// Dial dials a gRPC server. `ctx` can be used to set a timeout/deadline for Dial.
+// However, the max timeout is 20 seconds.
```

Comment:
So where does the 20 seconds come from? I thought signaling logic was giving us a 10 second hard upper limit.

Reply:
Right so the 20 seconds seems to come from the answerer side of things— also an RPC error

At 1:10 [here](https://www.loom.com/share/ab687976cd5d41bf83c9f6fc5e56ed4e?sid=5af55117-e749-4b70-b598-21831534d3c4), I get the error

Reply:
Gotcha; thanks for the video link!

Reply:
I moved it to right before the return and it still maxes out at 20